src/test/scala/chiselTests/QueueFlushSpec.scala

```diff
@@ -70,19 +70,28 @@ class QueueGetsFlushedTester (elements: Seq[Int], queueDepth: Int, bitWidth: Int
   q.io.enq.bits := elems(inCnt.value)
   when(q.io.enq.fire()) {
     inCnt.inc()
-    inCnt.inc()
+    currQCnt := currQCnt + 1.U //counts how many items have been enqueued
   }
   when(q.io.deq.fire()) {
-    outCnt.inc()
+    //ensure that what comes out is what comes in
+    assert(flushRegister === false.B)
+    assert(currQCnt <= queueDepth.U)
+    assert(elems(outCnt /*- 1.U*/) === q.io.deq.bits)
+    outCnt := outCnt + 1.U
+    when (currQCnt > 0.U) {
+      currQCnt := currQCnt - 1.U
+    }
   }
-
-  when(flushRegister) {
-    //check that queue gets flushed 
-    assert(!q.io.deq.valid, "Expected to not be able to dequeue a flushed queue because queue is empty") 
-    assert(q.io.enq.ready, "Expected enqueue to be ready when flush is high because queue is empty")
+  when(flush) {
+    outCnt := outCnt + Mux(q.io.enq.fire(), (currQCnt + 1.U), currQCnt)
+    currQCnt := 0.U //resets the number of items currently inside queue
+  }
+  when(flushRegister) { //Internal signal maybe_full is a register so some signals update on the next cycle
+    //check that queue gets flushed when queue is full
     assert(q.io.count === 0.U)
+    assert(!q.io.deq.valid, "Expected to not be able to dequeue a flushed queue if it had elements prior to the flush")
```

Comment:
maybe it would make more sense to move this to the `when(flush)` block of logic? I think we are expected to be able to dequeue on the same cycle that flush goes high.

Reply:
i think this check is still good though, it's basically ensuring the queue is empty. I would just change the message, and add another check above